src/language/dafnyInstallation.ts

```diff
@@ -1,477 +1,163 @@
-        + ' and ensure that the path containing javac is in the PATH environment variable. '
-        + 'You can obtain a free open-source JDK 1.8 from here: '
-        + 'https://aws.amazon.com/corretto/');
-        return false;
-      }
+  private async getDotnetToolVersion(): Promise<string> {
+    if(this.toolVersionCache === undefined) {
+      this.toolVersionCache = await DafnyInstaller.getDotnetToolVersionUncached();
     }
+    return this.toolVersionCache;
+  }
 
-    // Clone the right version
-    await this.execLog(`git clone -b v${configuredVersion} --depth 1 --recurse-submodules ${LanguageServerConstants.DafnyGitUrl}`);
-    processChdir(Utils.joinPath(installationPath, 'dafny').fsPath);
+  private static async getDotnetToolVersionUncached(): Promise<string> {
+    const { path: dotnetExecutable } = await getDotnetExecutablePath();
+    const { stdout } = await execFileAsync(dotnetExecutable, [ 'tool', 'search', 'Dafny', '--detail', '--prerelease' ]);
+    const entries = stdout.split('----------------').map(entry => entry.split('\n').filter(e => e !== ''));
+    const dafnyEntry = entries.filter(entry => entry[0] === 'dafny')[0];
+    const versions = dafnyEntry.slice(8).map(versionLine => versionLine.trimStart().split(' ')[0]);
 
-    const { path: dotnet } = await getDotnetExecutablePath();
-    // The DafnyCore.csproj has a few targets that call `dotnet` directly.
-    // If dotnet is configured in dafny.dotnetExecutablePath
-    // it MAY NOT be on the path.
-    // This will cause the build to fail.
-    // This works around this edge case.
-    const injectPath = `PATH=${path.dirname(dotnet)}:$PATH`;
-    // Build the DafnyLanguageServer
-    await this.execLog(`${injectPath} ${ (await getDotnetExecutablePath()).path } build Source/DafnyLanguageServer/DafnyLanguageServer.csproj`);
-    const binaries = Utils.joinPath(installationPath, 'dafny', 'Binaries').fsPath;
-    processChdir(binaries);
-    try {
-      await this.execLog('brew update'); // Could help some users not get "Error: The `brew link` step did not complete successfully"
-    } catch(error: unknown) {
-      this.writeStatus(`Could not run \`brew update\` but this step is optional (${error})`);
+    const versionDescription = process.env['dafnyIdeVersion'] ?? Configuration.get<string>(ConfigurationConstants.PreferredVersion);
+    let toolVersion: string;
+    switch(versionDescription) {
+    case LanguageServerConstants.LatestStable: {
+      const version = LanguageServerConstants.LatestVersion;
+      toolVersion = versions.filter(l => l.startsWith(version))[0];
+      window.showInformationMessage(`Using latest stable version: ${toolVersion}`);
+      break;
+    }
+    case LanguageServerConstants.LatestNightly: {
+      const nightlies = versions.filter(l => l.includes('nightly'));
+      const dates: { index: number, date: string }[] = nightlies.map((n, index) => {
+        const split: string[] = n.split('-');
```

Comment:
Could you please add a comment about what to expect here, a typical string that is being splitted?